README.md

```diff
@@ -129,19 +138,26 @@ var builder = new HttpRequestInterceptionBuilder()
 var client = options.CreateHttpClient();
 
 // Throws an HttpRequestException
-await client.GetStringAsync("http://public.je-apis.com");
+await Assert.ThrowsAsync<HttpRequestException>(
+    () => client.GetStringAsync("http://public.je-apis.com"));
 ```
+<sup><a href='/tests/HttpClientInterception.Tests/Examples.cs#L29-L45' title='Snippet source file'>snippet source</a> | <a href='#snippet-fault-injection' title='Start of snippet'>anchor</a></sup>
+<!-- endSnippet -->
 
 #### Registering Request Interception When Using IHttpClientFactory
 
 If you are using [`IHttpClientFactory`](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests "Use IHttpClientFactory to implement resilient HTTP requests") to register `HttpClient` for Dependency Injection in a .NET Core 2.1 application (or later), you can implement a custom [`IHttpMessageHandlerBuilderFilter`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.http.ihttpmessagehandlerbuilderfilter "IHttpMessageHandlerBuilderFilter Interface") to register during test setup, which makes an instance of `HttpClientInterceptorOptions` available to inject an HTTP handler.
 
 A working example of this approach can be found in the [sample application](https://github.com/justeat/httpclient-interception/blob/main/samples/README.md "Sample application that uses JustEat.HttpClientInterception").
 
-```csharp
-using Microsoft.Extensions.Http;
-
-public class InterceptionFilter : IHttpMessageHandlerBuilderFilter
+<!-- snippet: interception-filter -->
+<a id='snippet-interception-filter'></a>
+```cs
+/// <summary>
+/// A class that registers an intercepting HTTP message handler at the end of
+/// the message handler pipeline when an <see cref="HttpClient"/> is created.
+/// </summary>
+private sealed class InterceptionFilter : IHttpMessageHandlerBuilderFilter
```

Comment:
I'd refactor the samples slightly so it comes out here as a `public` class, as seems a bit weird to have the example showing it as a `private` class without seeing the context that it's nested in a fixture.

Reply:
that causes 

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CA1034	Do not nest type InterceptionFilter. Alternatively, change its accessibility so that it is not externally visible.	SampleApp.Tests (net5.0)	C:\Code\httpclient-interception\samples\SampleApp.Tests\HttpServerFixture.cs	60	Active
```

Reply:
Yeah that's why I meant refactor, rather than just change the accessibility modifier. I would change the sample to not have it nested so it gave a nicer example snippet.